hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -853,7 +852,7 @@ object HoodieBaseRelation extends SparkAdapterSupport {
       val hoodieConfig = new HoodieConfig()
       hoodieConfig.setValue(USE_NATIVE_HFILE_READER,
         options.getOrElse(USE_NATIVE_HFILE_READER.key(), USE_NATIVE_HFILE_READER.defaultValue().toString))
-      val reader = HoodieFileReaderFactory.getReaderFactory(HoodieRecordType.AVRO)
+      val reader = (new HoodieSparkIOFactory).getReaderFactory(HoodieRecordType.AVRO)
```

Comment:
Similar here.

If the IO factory class name is already set in the storage config, could we use the reflection, i.e., `HoodieIOFactory.getIOFactory(conf)`, to load the `HoodieSparkIOFactory`, which achieves the same behavior?

Reply:
Yeah, that would achieve the same thing. Do you think that would be better.

Reply:
Based on the discussion, it is safer to hardcode the class for now as there are gaps in passing the storage configuration outside the `hudi-common` module.